core/src/main/java/com/github/mizool/core/validation/CheckCountryCode.java

```diff
@@ -1,41 +1,53 @@
 /**
- *  Copyright 2017 incub8 Software Labs GmbH
- *  Copyright 2017 protel Hotelsoftware GmbH
+ * Copyright 2017 incub8 Software Labs GmbH
+ * Copyright 2017 protel Hotelsoftware GmbH
  *
- *  Licensed under the Apache License, Version 2.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
  *
- *      http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
  *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  */
 package com.github.mizool.core.validation;
 
 import java.util.Locale;
 import java.util.Set;
 
+import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
 
 import com.google.common.collect.ImmutableSet;
 
-public class CheckCountryCode extends AbstractTypeMappingValidator<CountryCode>
+public class CheckCountryCode implements ConstraintValidator<CountryCode, String>
 {
     private static final Set<String> COUNTRY_CODES = ImmutableSet.copyOf(Locale.getISOCountries());
 
+    private boolean mandatory;
+
     @Override
-    public void initialize(CountryCode notEmpty)
+    public void initialize(CountryCode countryCode)
     {
+        mandatory = countryCode.mandatory();
     }
 
     @Override
-    protected boolean isNonNullNonEmptyStringValid(
-        String validationObject, ConstraintValidatorContext constraintValidatorContext)
+    public boolean isValid(String validationObject, ConstraintValidatorContext constraintValidatorContext)
     {
-        return COUNTRY_CODES.contains(validationObject);
+        boolean valid = false;
```

Comment:
refactor this so `isValid()` only contains `return isNullButOptional(validationObject) || isValidValue(validationObject);`. do the same for the other checks.

this way, the reader does not have to parse a mixture of negations and if/else structure.